recipes/libcufile/meta.yaml

```diff
@@ -0,0 +1,107 @@
+{% set name = "libcufile" %}
+{% set version = "1.5.0.59" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set extension = "tar.xz" %}  # [not win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/{{ name }}/{{ platform }}/{{ name }}-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: 222c49b29afb544c69542c4e658b0a1bd01fd2306f2136ad6990ccfff482a097  # [linux64]
+  sha256: 4e8d2b525322be9ab6ae574bd1d0325a4bf9c6c3bbdaa1611b5b969b7b34b0c4  # [aarch64]
+  patches:
+    - Makefile.patch
+
+build:
+  number: 0
+  skip: true  # [osx]
+  skip: true  # [ppc64le]
+  skip: true  # [aarch64]
+  skip: true  # [win]
+
+outputs:
+  - name: libcufile
+    build:
+      missing_dso_whitelist:
+        - "*/libcuda.so.*"                     # [linux]
+    run_exports:
+      - {{ pin_subpackage("libcufile", max_pin="x") }}
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+      host:                                    # [linux]
+        - sysroot_{{ target_platform }} 2.17   # [linux]
+        - __glibc >=2.17                       # [linux]
+      run:                                     # [linux]
+        - sysroot_{{ target_platform }} 2.17   # [linux]
+        - __glibc >=2.17                       # [linux]
+    files:
+      - lib/libcufile*.so.*
+      - etc/cufile.json
+      - man
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcufile*.so.*
+        - test -f $PREFIX/etc/cufile.json
+
+  - name: libcufile-dev
+    run_exports:
+      - {{ pin_subpackage("libcufile", max_pin="x") }}
+    requirements:
+      run:
+        - libcufile >={{ version }}
+      run_contrained:
+        - libcufile-static >={{ version }}
+    files:
+      - lib/libcufile*.so
+      - targets/*/lib/libcufile*.so
+      - include/cufile*
+      - targets/*/include/cufile*
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcufile*.so
+        - test -f $PREFIX/include/cufile*
+
+  - name: libcufile-static
+    requirements:
+      host:
+        - libcufile-dev {{ version }}
+      run:
+        - {{ pin_subpackage("libcufile", max_pin="x") }}
+    files:
+      - lib/libcufile*_static.a
+      - targets/*/lib/libcufile*_static.a
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcufile*_static.a
+
+  - name: gds-tools
+    requirements:
+      host:
+        - libcufile-dev {{ version }}
+      run:
+        - {{ pin_subpackage("libcufile", max_pin="x") }}
```

Comment:
It looks like we are seeing [linking errors on CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=656065&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1143 )

It appears this is related to the compiler runtime libraries. Adding the compilers as `build` dependency should fix this as the compiler libraries will then be added & picked up as `host` & `run` dependencies automatically

Also looks like [`libnuma` is needed, but not found]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=656065&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1179 ). So suggesting adding this to `host`. As [`libnuma` has `run_exports`]( https://github.com/conda-forge/numactl-feedstock/blob/250059ce5f4997c510edba768524cf3b2ab57834/recipe/meta.yaml#L14-L15 ), it will automatically added itself to `run` dependencies. This should handle this CI error.

```suggestion
    requirements:
      build:
        - {{ compiler("c") }}
        - {{ compiler("cxx") }}
      host:                                    # [linux]
        - sysroot_{{ target_platform }} 2.17   # [linux]
        - __glibc >=2.17                       # [linux]
        - libcufile-dev {{ version }}
        - libnuma                              # [linux]
      run:                                     # [linux]
        - sysroot_{{ target_platform }} 2.17   # [linux]
        - __glibc >=2.17                       # [linux]
        - {{ pin_subpackage("libcufile", max_pin="x") }}
```